### ISSUE TITLE
Extreme OS (xos) don't match prompt when config was changed and don't saved

### ISSUE THREAD

**Issue by glaubway** [ytti/oxidized#2113]
Hello,
have a problem when config at switch was changed and didn't save. Prompt have "* " before hostname.
```
* sw0.l9.kv.1 # disable clipaging session
                                  ^
%% Invalid input detected at '^' marker.
* sw0.l9.kv.2 # 
```
and that prompt don't match with post_login.
```
    post_login do
      data = cmd 'disable clipaging session'
      match = data.match /^disable clipaging session[\n\r]\*?\s?[-\w]+\s?[-\w.~]+(:\d+)? [#>] $/m
      next if match

      cmd 'disable clipaging'
    end
```
If config saved, oxidized work perfectly.
```
sw0.l9.kv.1 # disable clipaging session
                                ^
%% Invalid input detected at '^' marker.
sw0.l9.kv.2 # disable clipaging
```

**Comment by DarkCatapulter** [comment]
I mentioned this issue in #2107 and have also proposed a fix for it. You can apply it manually if you don't feel like waiting for a pull request.

**Comment by DarkCatapulter** [comment]
I see that you followed my suggestion in #2107. Try using this instead: `/^disable clipaging session\n\r?\*?\s?[-\w]+\s?[-\w.~]+(:\d+)? [#>] $/m`

**Comment by glaubway** [comment]
> I see that you followed my suggestion in #2107. Try using this instead: `/^disable clipaging session\n\r?\*?\s?[-\w]+\s?[-\w.~]+(:\d+)? [#>] $/m`.
> 
> And if that doesn't work, try replacing the `\r?` with `\s?`. Please do tell if it works or not :)

Hi, trying both variant, but no. :(

I saw in logs that oxi stoped at this moment:
```
D, [2020-06-04T09:52:53.046128 #38401] DEBUG -- : lib/oxidized/input/cli.rb: Running post_login command: nil, block: #<Proc:0x0000562564821290@/var/lib/gems/2.5.0/gems/oxidized-0.28.0/lib/oxidized/model/xos.rb:44> at sw0.l9.kv
D, [2020-06-04T09:52:53.046344 #38401] DEBUG -- : lib/oxidized/input/ssh.rb disable clipaging session @ sw0.l9.kv with expect: /^\*?\s?[-\w.~]+(:\d+)? [#>] $/
D, [2020-06-04T09:52:53.046752 #38401] DEBUG -- socket[3fb78612a7f0]: queueing packet nr 11 type 94 len 44
D, [2020-06-04T09:52:53.046895 #38401] DEBUG -- : lib/oxidized/input/ssh.rb: expecting [/^\*?\s?[-\w.~]+(:\d+)? [#>] $/] at sw0.l9.kv
```
Debug output:
```
Press the <tab> or '?' key at any time for completions.
Remember to save your configuration changes.


* sw0.l9.kv.1 # disable clipaging session
                                  ^
%% Invalid input detected at '^' marker.
* sw0.l9.kv.2 #
```
I really can't understand why oxi can't parse prompt. When i check regexp with my prompt at online checker, it's match fully.

**Comment by DarkCatapulter** [comment]
@glaubway It should work as you state, and I did troubleshoot this in #2107. In my latest comment in that issue,  I state that there appears to be a bug with some code that is supposed to strip `\r` and whitespaces. xos inserts leading \r characters and other trailing white space, and this is causing an issue with the regexp.

Did you only change the regexp for "Disable clipaging session" or did you also change for the prompt? You should change the prompt to `/^\s?\*?\s?[-\w]+\s?[-\w.~]+(:\d+)? [#>] $/` as well. I thought you had already done this, but looking at your debug statement it appears you haven't changed it.

**Comment by glaubway** [comment]
@DarkCatapulter I update my previous comment, after change prompt it's work.

**Comment by DarkCatapulter** [comment]
@glaubway Great! Happy that this resolved your issue 😄
You're welcome :)

**Comment by wk** [comment]
Presumed closed by #2117
